syntax = "proto3";

service ServidorMSN{
  rpc RecibirMensaje (Usuario) returns (stream Mensaje);
  rpc EnviarMensaje(Mensaje) returns (Ack);
  rpc DevolverUsuarios (Usuario) returns (Usuario);
  rpc ValidarUsuario (Usuario) returns (Ack);
  rpc DevolverMensajes (Usuario) returns (stream Mensaje);
  rpc Desconectarse (Usuario) returns (Ack);

}

message Usuario {
  string username = 1;
}

message Mensaje {
  string id = 1;
  string emisor = 2;
  string receptor = 3;
  string mensaje = 4;
  string marcatiempo = 5;
}

message Ack{
    string ack = 1;
}
